rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -536,6 +554,112 @@ void saveOldFederation_after_RSKIP123_should_save_null() {
+
+        when(repository.getStorageBytes(
+            bridgeAddress,
+            NEW_FEDERATION_BTC_UTXOS_KEY_FOR_TESTNET_PRE_HOP.getKey()
+        )).thenReturn(BridgeSerializationUtils.serializeUTXOList(federationUtxosAfterRskip284Activation));
+
+        List<UTXO> federationUtxosAfterRskip293Activation = BitcoinTestUtils.createUTXOs(2, btcAddress);
+        when(repository.getStorageBytes(
+            bridgeAddress,
+            NEW_FEDERATION_BTC_UTXOS_KEY_FOR_TESTNET_POST_HOP.getKey()
+        )).thenReturn(BridgeSerializationUtils.serializeUTXOList(federationUtxosAfterRskip293Activation));
+
+        FederationStorageProvider federationStorageProvider = createFederationStorageProvider(repository);
+
+        List<UTXO> obtainedUtxos = federationStorageProvider.getNewFederationBtcUTXOs(networkParameters, activations);
+
+        if (!networkId.equals(NetworkParameters.ID_TESTNET)) {
+            Assertions.assertEquals(federationUtxos, obtainedUtxos);
+            return;
+        }
+
+        // testnet
+        // rskip284 & rskip293 are not active
+        if (!isRskip284Active) {
+            Assertions.assertEquals(federationUtxos, obtainedUtxos);
+            return;
+        }
+
+        // rskip284 is active
+        if (!isRskip293Active) {
+            Assertions.assertEquals(federationUtxosAfterRskip284Activation, obtainedUtxos);
+            return;
+        }
+
+        // rskip293 is active
+        Assertions.assertEquals(federationUtxosAfterRskip293Activation, obtainedUtxos);
+    }
+
+    @Test
+    void getNewFederationBtcUTXOs_calledTwice_returnCachedUtxos() {
+
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.only(ConsensusRule.RSKIP123).forBlock(0);
+        StorageAccessor storageAccessor = new InMemoryStorage();
+
+        DataWord newFederationBtcUtxosKey = NEW_FEDERATION_BTC_UTXOS_KEY.getKey();
+
+        Address btcAddress = new Address(testnetBtcParams, Hex.decode("4a22c3c4cbb31e4d03b15550636762bda0baf85a"));
+        List<UTXO> expectedUtxos = BitcoinTestUtils.createUTXOs(1, btcAddress);
+        storageAccessor.saveToRepository(newFederationBtcUtxosKey, expectedUtxos, BridgeSerializationUtils::serializeUTXOList);
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        List<UTXO> actualUtxos = federationStorageProvider.getNewFederationBtcUTXOs(testnetBtcParams, activations);
+
+        assertEquals(1, actualUtxos.size());
+        assertEquals(expectedUtxos, actualUtxos);
+
+        List<UTXO> extraUtxos = new ArrayList<>(expectedUtxos);
+        extraUtxos.addAll(BitcoinTestUtils.createUTXOs(1, btcAddress));
+
+        storageAccessor.saveToRepository(newFederationBtcUtxosKey, extraUtxos, BridgeSerializationUtils::serializeUTXOList);
+
+        List<UTXO> actualUtxosAfterSecondGet = federationStorageProvider.getNewFederationBtcUTXOs(null, activations);
```

Comment:
Why pass `null` as the network parameters?

Reply:
Because it won't be used, and this helps ensure and confirm that it will not get used, since it shouldn't get used this time, it should simply return the cached list of utxos.